src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/typeresolution.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <RoMetadataApi.h>
+#include <hstring.h>
+#include <atlbase.h>
+#include <windows.foundation.h>
+#include <Windows.h>
+#include "wil/result.h"
+#include "wil/resource.h"
+#include "wil/filesystem.h"
+#include "Synchapi.h"
+#include <unordered_map>
+
+#define METADATA_FILE_EXTENSION L"winmd"
+#define METADATA_FILE_PATH_FORMAT L"%s%s."  METADATA_FILE_EXTENSION
+#define METADATA_FILE_SEARCH_FORMAT L"%s%s*."  METADATA_FILE_EXTENSION
+
+namespace UndockedRegFreeWinRT
+{
+    // This number comes from (c_cchTypeNameMax = 512) in //depot/fbl_ie_dev1/com/WinRT/buildtools/mdmerge/MetadataScope.h
```

Comment:
This looks like some pretty inside information. Probably shouldn't do that.

Reply:
Oh yea let me comb through the headers to make sure